salsah/src/public/js/jquery.htmleditor.js

```diff
@@ -138,13 +138,18 @@
 
                 // specify allowed elements, attributes, and classes
                 // this must conform to the `STANDARD_MAPPING`
-                var filter = ' p em strong strike u sub sup; a[!href](salsah-link)';
+                var filter = ' p em strong strike u sub sup hr h1 h2 h3 h4 h5 h6 pre table tbody tr td ol ul li cite blockquote code; a[!href](salsah-link) ';
```

Comment:
As for the filter rules, you could provide this link: https://docs.ckeditor.com/#!/guide/dev_advanced_content_filter-section-custom-mode-example